program/src/processor.rs

```diff
@@ -5987,6 +5987,184 @@ impl Processor {
             referrer_id,
         )
     }
+
+    #[inline(never)]
+    fn cancel_all_spot_orders(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        limit: u8,
+    ) -> MangoResult {
+        const NUM_FIXED: usize = 21;
+
+        let [
+            mango_group_ai, // read
+            mango_cache_ai,   // read
+            mango_account_ai, // write
+            owner_ai, // read, signer
+            base_root_bank_ai,      // read
+            base_node_bank_ai,      // write
+            base_vault_ai,          // write
+            quote_root_bank_ai,     // read
+            quote_node_bank_ai,     // write
+            quote_vault_ai,         // write
+
+            spot_market_ai,         // write
+            bids_ai,                // write
+            asks_ai,                // write
+            open_orders_ai,         // write
+            signer_ai,              // read
+            dex_event_queue_ai,     // write
+            dex_base_ai,            // write
+            dex_quote_ai,           // write
+            dex_signer_ai,          // read
+            dex_prog_ai,            // read
+            token_prog_ai,          // read
+        ] = array_ref![accounts, 0, NUM_FIXED];
```

Comment:
force_cancel_spot_orders has a
```
check_eq!(token_prog_ai.key, &spl_token::ID, MangoErrorCode::InvalidProgramId)?;
```
here, but I think that's unnecessary since serum will check it again. I'd either add it here or drop the one in force_cancel (@dafyddd ?)

Reply:
yeah I would just add it.